tempel.el

```diff
@@ -576,11 +578,23 @@ This is meant to be a source in `tempel-template-sources'."
             (delq (assq-delete-all 'tempel--active minor-mode-overriding-map-alist)
                   minor-mode-overriding-map-alist)))))
 
-(defun tempel-done ()
-  "Template completion is done."
+(defun tempel-done (&optional st)
```

Comment:
Please don't make the argument part of the public API. The template st states are nowhere part of the public API. Instead introduce a helper function `tempel--done` (or with some better name) which will execute the callback and then call `tempel--disable`.